.github/workflows/ci.yml

```diff
@@ -0,0 +1,267 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  linters:
+    strategy:
+      matrix:
+        go-version: [ 1.18, 1.19 ]
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: ${{ matrix.go-version }}
+
+    - name: Check vendored dependencies
+      run: |
+        go mod tidy
+        go mod verify
+        go mod vendor -v
+        git status
+        if [[ -n "$(git status --porcelain vendor)" ]]; then
+          echo 'vendor/ is out-of-date: run `go mod tidy && go mod vendor` and then check in the changes'
+          echo 'If `go mod tidy && go mod vendor` results in no changes, make sure you are using the latest relase of Go'
+          git status --porcelain vendor
+          exit 1
+        fi
+
+    - name: Golangci-lint
```

Comment:
Best practice is to make this its own workflow, so it runs in parallel and you get early feedback.

also, please set `version: latest` and `only-new-issues: true` in the `with` section so that PRs aren't overwhelmed with issues

Reply:
(I was just experimenting with linters this weekend in uio https://github.com/u-root/uio/blob/main/.github/workflows/golangci-lint.yml)